crypto/txsigner.go

```diff
@@ -228,15 +228,15 @@ func (e *EIP155Signer) CalculateV(parity byte) []byte {
 }
 
 // encodeSignature generates a signature value based on the R, S and V value
-func encodeSignature(R, S *big.Int, V byte) ([]byte, error) {
+func encodeSignature(R, S, V *big.Int) ([]byte, error) {
 	if !ValidateSignatureValues(V, R, S) {
 		return nil, fmt.Errorf("invalid txn signature")
 	}
 
 	sig := make([]byte, 65)
 	copy(sig[32-len(R.Bytes()):32], R.Bytes())
 	copy(sig[64-len(S.Bytes()):64], S.Bytes())
-	sig[64] = V
+	sig[64] = byte(V.Int64()) // here is safe to convert it since ValidateSignatureValues will validate the v value
```

Comment:
Not sure myself about this proposal, but think it is more consistent and offering an alternative. However you wish.

```suggestion
sig[64] = V.Bytes()[0] // here is safe to convert it since ValidateSignatureValues will validate the v value
```

Reply:
I would leave as it is for now, since it is safe to convert it, because of the validation we perform before this.